docs/manual/config/keys.rst

```diff
@@ -45,6 +45,24 @@ The :class:`EzKey` modifier keys (i.e. ``MASC``) can be overwritten through the
        'C': 'control',
     }
 
+Callbacks can also be configured to work only under certain conditions by using
+the ``when()`` method. Currently, two conditions are supported:
+
+::  
+
+    from libqtile.config import Key
+
+    def myfunc():
+        pass
+
+    keys = [
+        # Only trigger callback for a specific layout
+        Key([mod], "a", lazy.function(myfunc).when(layout='max'))
+
+        # When the current window is floating (default True)
+        Key([mod], "b", lazy.function(myfunc).when(when_floating=False))
```

Comment:
Could we instead use other `lazy` paths, and maybe two different ones? From seeing other people's configs I get the impression `lazy.function` is less widely used than e.g. `lazy.layout.next`, and having the `def` draws attention to that. With two different ones, it is more clear that `when` is applicable not just to that example.